crypto/src/main/java/net/corda/v5/crypto/merkle/MerkleProof.java

```diff
@@ -15,37 +18,39 @@
  * @property treeSize The total number of leaves in the tree.
  * @property leaves The leaf items whose inclusion is proved by the proof.
  * @property hashes The helper hashes needed to reconstruct the whole tree.
- *
  */
 @CordaSerializable
-interface MerkleProof {
-    val proofType: MerkleProofType
-    val treeSize: Int
-    val leaves: List<IndexedMerkleLeaf>
-    val hashes: List<SecureHash>
+public interface MerkleProof {
+
+    /**
+     * Get the type of the proof
+     *
+     * @return Type of the proof
+     */
+    @NotNull
+    MerkleProofType getProofType();
+
+    int getTreeSize();
+    
+    List<IndexedMerkleLeaf> getLeaves();
+
+    List<SecureHash> getHashes();
 
     /**
      * Checks if the [MerkleProof] has been generated from a [MerkleTree] with the given [root].
-     * @param root The root of the tree to be verified.
-     * @param digest The tree's digest.
      *
+     * @param root   The root of the tree to be verified.
+     * @param digest The tree's digest.
      * @returns Result of the verification.
      */
-    fun verify(
-        root: SecureHash,
-        digest: MerkleTreeHashDigest
-    ): Boolean
+    boolean verify(@NotNull SecureHash root, @NotNull MerkleTreeHashDigest digest);
 
     /**
      * Rebuilds the [MerkleTree] from the [MerkleProof] and returns its root [SecureHash].
      *
      * @param digest The tree's digest.
-     *
-     * @returns Root hash of the tree.
-     *
      * @throws MerkleProofRebuildFailureException if the calculation of the root hash failed.
+     * @returns Root hash of the tree.
      */
-    fun calculateRoot(
-        digest: MerkleTreeHashDigest
-    ): SecureHash
+    SecureHash calculateRoot(@NotNull MerkleTreeHashDigest var1);
```

Comment:
Any reason why did you use `var1` instead of `digest`? Just wondering.

Reply:
Good question! That's what the bytecode decompiler generates, and I went too fast and missed that when cut and pasting. Definetly don't want to call it var1 in the final change. @kyriathar has the baton at the moment.

Reply:
fixed